SherpaFrontEnd.Tests/Acceptance/CookiesComplianceAcceptanceTest.cs

```diff
@@ -0,0 +1,92 @@
+    private const string CookiesAcceptedDate = "CookiesAcceptedDate";
+    private readonly TestContext _testCtx;
+    private ILocalStorageService _mockLocalStorage;
+
+    public CookiesComplianceAcceptanceTest()
+    {
+        _testCtx = new TestContext();
+        _testCtx.JSInterop.SetupVoid(FunctionName).SetVoidResult();
+        _testCtx.Services.AddScoped<ICookiesService, CookiesService>();
+        _mockLocalStorage = _testCtx.AddBlazoredLocalStorage();
+    }
+
+
+    [Fact]
+    public async Task ShouldLetTheUserAcceptCookiesIfTheyHaveNotBeenApprovedYet()
+    {
+        var app = _testCtx.RenderComponent<CookiesPopup>();
+
+        var popupElement = app.FindElementByCssSelectorAndTextContent("div[role=\"dialog\"]", "To find out more about our Cookies policy here. Once you are done, please, come back and accept them.");
+        
+        var acceptedLocalStorageEntry = await GetAcceptedLocalStorageEntry();
+        Assert.Null(acceptedLocalStorageEntry);
+ 
+        AssertPopupIsShown(popupElement);
+
+        var approveButton = app.FindElementByCssSelectorAndTextContent("button", "Accept");
+        approveButton!.Click();
+
+        acceptedLocalStorageEntry = await _mockLocalStorage.GetItemAsync<long?>(CookiesAcceptedDate);
+        Assert.NotNull(acceptedLocalStorageEntry);
+        
+        popupElement = app.FindElementByCssSelectorAndTextContent("div[role=\"dialog\"]", "To find out more about our Cookies policy here. Once you are done, please, come back and accept them.");
+        AssertPopupIsNotShown(popupElement);
+        _testCtx.JSInterop.VerifyInvoke(FunctionName);
+    }
+
+    [Fact]
+    public async Task ShouldLetTheUserAcceptCookiesIfTheyHaveBeenApprovedMoreThanSixMonthsAgo()
+    {
+        var sevenMonthsAgo = DateTimeOffset.Now.AddMonths(-7).ToUnixTimeMilliseconds();
+        await _mockLocalStorage.SetItemAsync(CookiesAcceptedDate, sevenMonthsAgo);
+        
+        var app = _testCtx.RenderComponent<CookiesPopup>();
+
+        var popupElement = app.FindElementByCssSelectorAndTextContent("div[role=\"dialog\"]", "To find out more about our Cookies policy here. Once you are done, please, come back and accept them.");
+ 
+        AssertPopupIsShown(popupElement);
+
+        var approveButton = app.FindElementByCssSelectorAndTextContent("button", "Accept");
+        approveButton!.Click();
+
+        var acceptedLocalStorageEntry = await _mockLocalStorage.GetItemAsync<long?>(CookiesAcceptedDate);
+        Assert.NotNull(acceptedLocalStorageEntry);
+        Assert.NotEqual(sevenMonthsAgo, acceptedLocalStorageEntry);
+        
+        popupElement = app.FindElementByCssSelectorAndTextContent("div[role=\"dialog\"]", "To find out more about our Cookies policy here. Once you are done, please, come back and accept them.");
+        AssertPopupIsNotShown(popupElement);
+        _testCtx.JSInterop.VerifyInvoke(FunctionName);
+    }
+
+    private static void AssertPopupIsShown(IElement? cookiesPopup)
```

Comment:
I like these asserts, I think this is something we should consider doing more often or refacoring in older test suites